components/multi-attribute-login/org.wso2.carbon.identity.unique.claim.mgt/src/main/java/org/wso2/carbon/identity/unique/claim/mgt/listener/UniqueClaimUserOperationEventListener.java

```diff
@@ -301,15 +317,20 @@ private boolean isClaimDuplicated(String username, String claimUri, String claim
             userList = userStoreMgrFromRealm.getUserList(claimUri, claimValue, profile);
         }
 
-        if (userList.length == 1) {
-            String usernameWithUserStoreDomain = UserCoreUtil.addDomainToName(username, domainName);
+        String usernameWithUserStoreDomain = UserCoreUtil.addDomainToName(username, domainName);
+        if (userList.length == 0) {
+            return false;
+        } else {
             if (usernameWithUserStoreDomain.equalsIgnoreCase(userList[0])) {
+                if (userList.length > 1 && log.isDebugEnabled()) {
+                    log.debug("Multiple users found for claim URI: " + claimUri + " with value: " +
+                            claimValue + ". The current user is the first in the list; skipping uniqueness " +
```

Comment:
Don't we need to mask the claim value here?

Reply:
This is fixed.